src/sentry/utils/registry.py

```diff
@@ -15,9 +15,10 @@ class NoRegistrationExistsError(ValueError):
 
 
 class Registry(Generic[T]):
-    def __init__(self):
+    def __init__(self, enable_reverse_lookup=True):
         self.registrations: dict[str, T] = {}
         self.reverse_lookup: dict[T, str] = {}
+        self.enable_reverse_lookup = enable_reverse_lookup
 
     def register(self, key: str):
```

Comment:
Separately, it could also be worth changing the way we register so that we support both a decorator and a function. Not a huge priority though.